src/client/util/http.cc

```diff
@@ -190,15 +191,46 @@ bool HTTP::DownloadViaClearnet() {
           for (auto const& header : response.headers()) {
             if (header.first == "ETag") {
               if (header.second != GetPreviousETag())
-                SetETag(header.second);  // Set new ETag
+                new_etag = header.second;
             }
             if (header.first == "Last-Modified") {
               if (header.second != GetPreviousLastModified())
-                SetLastModified(header.second);  // Set new Last-Modified
+                new_last_modified = header.second;
             }
+            if (header.first == "Content-Length")
+              {
+                // Convert Content-Length string to LengthLimit comparable type
+                auto clen =
+                    boost::lexical_cast<std::underlying_type_t<LengthLimit>>(
+                        header.second);
```

Comment:
>                // Convert Content-Length string to LengthLimit comparable type
>                auto clen =
>                    boost::lexical_cast<std::underlying_type_t<LengthLimit>>(
>                        header.second);

This is overkill. Simply do:

```cpp
std::uint64_t const clen = std::stoull(header.second);
```

Reply:
Thought it might be, but wanted to make flexible for possible changes to `LengthLimit`'s type. Will make the recommended change.